firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryEngine.java

```diff
@@ -82,12 +82,13 @@ public ImmutableSortedMap<DocumentKey, Document> getDocumentsMatchingQuery(
       return result;
     }
 
-    result = performQueryUsingRemoteKeys(query, remoteKeys, lastLimboFreeSnapshotVersion);
+    AutoIndexing counter = new AutoIndexing();
+    result = performQueryUsingRemoteKeys(query, remoteKeys, lastLimboFreeSnapshotVersion, counter);
```

Comment:
Here `counter` can be `@Nullable` to indicate index-free queries do no collect information.

Reply:
doesn't `index-free queries` do a full collection scan, which collects infomation?

Reply:
Index free does not do full collection scan. It takes the result set from backend, and amend it with the local mutations and documents with newer version than the result set. It does not accurately reflect how much a full collection scan will cost.